### DISCUSSION TITLE
Hide next/prev button on first/last slide respectively

### DISCUSSION THREAD

**Discussion by BenHappyPlates** [nolimits4web/swiper#4348] [Q&A]
Good Day Ladies and Gentleman, 

i'm using NextJS und Hooks and I'm trying to hide the navigation arrows if the active slide is the first or the last item in the array. 

i read that there is a disabledClass and a hiddenClass but i'm not sure how to use them correctly

i'd really appreciate any advice 

regards Ben

**Comment by jpierson-at-riis** [comment]
I'm trying to do this with the following style rule but it doesn't seem to be working for me at the moment. 

```css
swiper-container::part(button-prev)[aria-disabled=true],
swiper-container::part(button-next)[aria-disabled=true] {
  display: none;
}
```

See my [code sandbox example](https://codesandbox.io/p/sandbox/xprr5y?file=%2Findex.html&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522cluwqdi8s0007356euol7w673%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522cluwqdi8s0003356ejh4383kr%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522cluwqdi8s0004356ex7g6ub13%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522cluwqdi8s0006356e6sqn1kum%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522cluwqdi8s0003356ejh4383kr%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cluwqdi8s0002356euu27uois%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Findex.html%2522%257D%255D%252C%2522id%2522%253A%2522cluwqdi8s0003356ejh4383kr%2522%252C%2522activeTabId%2522%253A%2522cluwqdi8s0002356euu27uois%2522%257D%252C%2522cluwqdi8s0006356e6sqn1kum%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cluwqdi8s0005356eoah8q0fs%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522cluwqdi8s0006356e6sqn1kum%2522%252C%2522activeTabId%2522%253A%2522cluwqdi8s0005356eoah8q0fs%2522%257D%252C%2522cluwqdi8s0004356ex7g6ub13%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522cluwqdi8s0004356ex7g6ub13%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D).

**Reply by jpierson-at-riis** [reply]
It seems that psudoselectors should be usable on ::part [pseudo-element](https://developer.mozilla.org/en-US/docs/Web/CSS/Pseudo-elements) so I'm not sure why this is not working.

https://developer.mozilla.org/en-US/docs/Web/CSS/::part

I've also tried the following class selector looking for `.swiper-button-disabled` but to no avail.

```css
swiper-container::part(button-prev).swiper-button-disabled,
swiper-container::part(button-next).swiper-button-disabled {
  display: none;
}
```

**Reply by jpierson-at-riis** [reply]
I'm reading up on this now. This SO answer seems to be speaking directly to the concepts here.

[StackOverflow - Can the CSS :part pseudo-selector be used to style nested web components?](https://stackoverflow.com/questions/71047699/can-the-css-part-pseudo-selector-be-used-to-style-nested-web-components)

[Explainer: CSS Shadow ::part and ::theme](https://github.com/fergald/docs/blob/master/explainers/css-shadow-parts-1.md)

**Reply by jpierson-at-riis** [reply]
Currently, the most straight forward approach that I can think of with what I know would be to [inject styles](https://swiperjs.com/element#injecting-styles) to override the internal styles of this part but it's possible that [new features](https://github.com/nolimits4web/swiper/issues/7467) could be added to make this easier to achieve.

**Comment by Baribj** [comment]
Since there is still no clean global solution for this, and for folks using the web components, you can do:
```html
<swiper-container
        navigation="true"
        direction="horizontal"
        threshold="10"
        slides-per-view="auto"
        space-between="15"
        slides-per-group-auto="true"
        inject-styles='[".swiper-button-disabled{ display:none !important; }"]'
      >
        .
        .
        .
</swiper-container>
```
